### PR TITLE
Add healthchecks

### LINKED ISSUE #5: Implement a proper liveness and readiness probe endpoints
Can we make sure we have a proper endpoint for liveness and readiness probes ? 

In particular let's make sure that the readiness probe is configurable so that we can define an ojbect to pull from a given bucket to test full working capabilities of the proxy

### PR DESCRIPTION
**Pull request by bcbrockway**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
-/vendor/
-/bin/
-/dist/
\ No newline at end of file
+/vendor
+/bin
+/dist
+gcsproxy
```
Comment (nabadger) [inline .gitignore]: it's moaning about newline EOF

File: main.go [hunk 2]
```diff
@@ -164,9 +160,36 @@ func proxy(w http.ResponseWriter, r *http.Request) {
...
+}
 
+// readinessProbeHandler attempts to get the metadata from each of a list of Google buckets provided as an argument via
+// --readiness-buckets. If it fails to retrieve metadata from ALL buckets provided the handler will return a 503
+// response. The default list of buckets provided contains the three public Google buckets (see
+// https://cloud.google.com/storage/docs/public-datasets).
+func readinessProbeHandler(w http.ResponseWriter, r *http.Request) {
+	var err error
+	for _, bucket := range strings.Split(*readinessBuckets, ",") {
+		_, err = client.Bucket(bucket).Attrs(ctx)
```
Comment (primeroz) [inline main.go]: should we parallelize this ?

what if there is a timeout to the first bucket and so it takes X number of seconds to return and fail everything even though only the first bucket was having issues ?

We should be able to do it by using `goroutines` https://tour.golang.org/concurrency/1

Comment (bcbrockway) [reply main.go]: Yeah, that's a great shout actually. Will have a look now.

Comment (nabadger) [reply main.go]: May have missed this - but what's the use-case for testing multiple buckets anyway?

**Review by primeroz — APPROVED** [review]

**Pull request merged**